bin/test.bash

```diff
@@ -41,6 +41,43 @@ function usage {
   exit 255
 }
 
+# Define cleanup() function. This needs to be defined ASAP to catch SIGINT and SIGTERM at the earliest signs of trouble.
+cleanup() {
+  echo "Cleaning up..."
+
+  # Terminate the dev server if it's running
+  if [ -n "$DEV_GID" ]; then
+    echo "Terminating the dev server..."
+    kill -INT -- -"$DEV_GID"
+    echo "Dev server terminated."
+  fi
+
+  # Terminate the builder if it's running
+  if [ -n "$BUILDER_GID" ]; then
+    echo "Terminating the builder..."
+    kill -INT -- -"$BUILDER_GID"
+    echo "Builder terminated."
+  fi
+
+  # Terminate the preview server if it's running
+  if [ -n "$PREVIEW_GID" ]; then
+    echo "Terminating preview server..."
+    kill -INT -- -"$PREVIEW_GID"
+    echo "Preview server terminated."
+  fi
+
+  # Terminate the live builder if it's running
+  if [ -n "$LIVE_GID" ]; then
+    echo "Terminating the builder..."
+    kill -INT -- -"$LIVE_GID"
+    echo "Builder terminated."
+  fi
+  # Exit the script
+  exit 130
```

Comment:
This should not exit with `130` all of the time.  If Cypress is terminated normally and the script enters cleanup from line 343 then the exit status should be as it was set on 333 or 337.  So maybe pass it as a parameter to `cleanup` from 343?

Reply:
That would make sense, I'll modify cleanup with a parameter and define it as 130 as default but will be overriden by any passed through exit code.